CreateSnapshot/src/test/java/org/opensearch/migrations/TestCreateSnapshot.java

```diff
@@ -0,0 +1,115 @@
+    final byte[] payloadBytes = "Success".getBytes(StandardCharsets.UTF_8);
+    final Map<String, String> headers = Map.of(
+            "Content-Type",
+            "text/plain",
+            "Content-Length",
+            "" + payloadBytes.length
+    );
+
+    @Test
+    public void testRepoRegisterAndSnapshotCreateRequests() throws Exception {
+        var snapshotName = "my_snap";
+
+        ArrayList<Pair<FullHttpRequest, String>> capturedRequestList = new ArrayList<Pair<FullHttpRequest, String>>();
+        try (var destinationServer = SimpleNettyHttpServer.makeNettyServer(false,
+                Duration.ofMinutes(10),
+                fl -> {
+                    capturedRequestList.add(Pair.of(fl, fl.content().toString(StandardCharsets.UTF_8)));
+                    return new SimpleHttpResponse(headers, payloadBytes, "OK", 200);
+                }))
+        {
+            final var endpoint = destinationServer.localhostEndpoint().toString();
+
+            var sourceClient = new OpenSearchClient(ConnectionContextTestParams.builder()
+                    .host(endpoint)
+                    .insecure(true)
+                    .build()
+                    .toConnectionContext());
+            var snapshotCreator = new S3SnapshotCreator(
+                    snapshotName,
+                    sourceClient,
+                    "s3://new-bucket",
+                    "us-east-2",
+                    List.of(),
+                    snapshotContext.createSnapshotCreateContext()
+            );
+            SnapshotRunner.run(snapshotCreator);
+
+            FullHttpRequest registerRepoRequest = capturedRequestList.get(0).getLeft();
+            String registerRepoRequestContent = capturedRequestList.get(0).getRight();
+            FullHttpRequest createSnapshotRequest = capturedRequestList.get(1).getLeft();
+            String createSnapshotRequestContent = capturedRequestList.get(1).getRight();
+
+            assert registerRepoRequest.uri().equals("/_snapshot/migration_assistant_repo");
+            assert registerRepoRequest.method().toString().equals("PUT");
+            assert registerRepoRequestContent.equals("{\"type\":\"s3\",\"settings\":{\"bucket\":\"new-bucket\",\"region\":\"us-east-2\",\"base_path\":null}}");
+            assert createSnapshotRequest.uri().equals("/_snapshot/migration_assistant_repo/" + snapshotName);
+            assert createSnapshotRequest.method().toString().equals("PUT");
+            assert createSnapshotRequestContent.equals("{\"indices\":\"_all\",\"ignore_unavailable\":true,\"include_global_state\":true}");
+        }
+    }
+
+    @Test
+    public void testSnapshotCreateWithIndexAllowlist() throws Exception {
+        var snapshotName = "my_snap";
+        var indexAllowlist = List.of("allowed_index_1", "allowed_index_2");
+
+        final FullHttpRequest[] createSnapshotRequest = new FullHttpRequest[1];
+        final String[] createSnapshotRequestContent = {""};
```

Comment:
These arrays were IntelliJ's suggestion to fix the issue that you can't use a non-final in a lambda. Open to a better suggestion if this is not the right way to resolve it.

Reply:
Yeah - sorry.  This is a java-ism.  It makes me realize why the capture specifiers in C++ `[=foo,&bar]` really aren't that bad!

You can also do an AtomicReference/AtomicInteger, etc.  I usually do the AtomicReference thing because it seems more clear to me.  Our codebase isn't to the level of consistency where it matters, especially for test cases!

Reply:
an AtomicReference/AtomicString would make more sense here

Reply:
Okay, cool, thanks -- updated this to AtomicReferences